tests/test_griduniverse.py

```diff
@@ -114,9 +167,83 @@ def test_custom_griduniverse_from_text_file(self):
 
         self.assertTrue((step_no + 1) == len(actions_to_take) and done)
 
+    def test_lever_textworld_created_from_text_file_with_wrong_and_right_lever_metadata(self):
+        """
+        Test whether we can complete this griduniverse created from a text file
+        within textworld_map_files folder. This level contains levers so it also tests that the
+        functionality of levers works correctly. If anything changes anywhere, this test will fail.
+        """
+
+        env = GridUniverseEnv(textworld_fp='../core/envs/textworld_map_files/lever_level_2.txt')
+        env.render()
+
+        # Very particular path to take for agent to go to each lever and finally get to terminal state
+        actions_to_take = [env.action_descriptor_to_int['DOWN']] + [env.action_descriptor_to_int['RIGHT']] * 6 + \
+                          [env.action_descriptor_to_int['LEFT']] * 5 + [env.action_descriptor_to_int['DOWN']] * 5 + \
+                          [env.action_descriptor_to_int['RIGHT']] * 3 + [env.action_descriptor_to_int['UP']] * 2 + \
+                          [env.action_descriptor_to_int['RIGHT']] * 2 + [env.action_descriptor_to_int['LEFT']] * 2 + \
+                          [env.action_descriptor_to_int['DOWN']] * 2 + [env.action_descriptor_to_int['RIGHT']] * 2
+
+        for step_no, action in enumerate(actions_to_take):
+            # env.render()  # set mode='graphic' for pyglet render
+            observation, reward, done, info = env.step(action)
+
+            if done:
+                print("Episode finished after {} timesteps".format(step_no + 1))
+                break
+
+        self.assertTrue((step_no + 1) == len(actions_to_take) and done)
+
+    def test_lever_textworld_breaks_from_reading_wrong_lever_metadata_from_text_file(self):
+        """
+        Creates temporary file with correct map/grid text but wrong lever metadata (broken python dictionary in text) e.g. "{5: {"
+        Final case tests empty dictionary which should not crash.
+        """
+
+        file_path = '../core/envs/textworld_map_files/test_text_file_with_broken_metadata.txt'
+
+        # Test wrong format of metadata (list) breaks loading
+        with self.assertRaises(TypeError):
+            with open(file_path, 'w') as file:
+                file.write('oooo\n')
```

Comment:
Replace this and all other scenarios with text files that are parsed by the function instead of writing the strings line by line

Reply:
@fgtoralesch but these are temporary files, I didn't want to have extra files when i could just make temporary files?